common-app-lib/test/cmr/common_app/test/services/provider_cache_test.clj

```diff
@@ -0,0 +1,49 @@
+(ns cmr.common-app.test.services.provider-cache-test
+  (:require
+   [clojure.test :refer :all]
+   [cmr.common-app.services.provider-cache :as provider-cache]
+   [cmr.common.hash-cache :as hash-cache]
+   [cmr.common.time-keeper :as tk]
+   [cmr.redis-utils.test.test-util :as test-util]))
+
+(use-fixtures :each tk/freeze-resume-time-fixture)
+(use-fixtures :each test-util/embedded-redis-server-fixture)
+
+(def providers
+  '({:provider-id "PROV1" :short-name "PROV1" :cmr-only false :small false :consortiums "EOSDIS GEOSS"}
+    {:provider-id "PROV2" :short-name "PROV2" :cmr-only false :small false :consortiums "EOSDIS GEOSS"}))
+
+(def providers-cache-map
+  {"PROV1" {:provider-id "PROV1"
+            :short-name "PROV1"
+            :cmr-only false
+            :small false
+            :consortiums "EOSDIS GEOSS"}
+   "PROV2" {:provider-id "PROV2"
+            :short-name "PROV2"
+            :cmr-only false
+            :small false
+            :consortiums "EOSDIS GEOSS"}})
+
+(deftest validate-providers-exist-test
+  (let [cache-key provider-cache/cache-key
+        test-context {:system {:caches {cache-key (provider-cache/create-cache)}}}
+        cache (get-in test-context [:system :caches cache-key])]
+    (hash-cache/reset cache cache-key)
+    (testing "Cache is empty"
+      (is (nil? (hash-cache/get-map cache cache-key))))
+    (testing "Refreshing cache"
+      (is (= '(1 1) (provider-cache/refresh-provider-cache test-context providers))))
+    (testing "Cache has values"
+      (is (= providers-cache-map (hash-cache/get-map cache cache-key))))
+    (testing "Validate that a provider exists."
+      (is (= `("PROV1") (provider-cache/validate-providers-exist test-context '("PROV1")))))
+    (testing "Validate that a provider does not exist."
+      (is (thrown? clojure.lang.ExceptionInfo (provider-cache/validate-providers-exist test-context '("PROV3")))))))
+
+(deftest job-config-test
+  (testing "testing the provider cache refresh job"
+    (is (= {:job-type cmr.common_app.services.provider_cache.RefreshProviderCacheJob
+            :job-key "SomeJob"
+            :daily-at-hour-and-minute [07 00]}
```

Comment:
hard coded time should be defined as a const and in a common file with all other time constants.

Reply:
This will be fixed with https://bugs.earthdata.nasa.gov/browse/CMR-9553 in one configuration file.